### PR TITLE
7 event messsage pojo flatbuffer converter

### PR DESCRIPTION
**Pull request by DChambersSTFC**
Still WIP

### CODE REVIEW

**Comment by DChambersSTFC** [comment]
Look into detector vectors, and possibly starting detectors

File: EventMessageVisualisationStreamer/src/eventmessage/EventMessagePOJOToEventMessage.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * Main method.
+     * @param eventMessagePOJO The eventMessagePOJO to be converted.
+     * @return Event Message FLatBuffer Byte Array.
+     * @throws RuntimeException for invalid values.
+     */
+    public static byte[] convert(final EventMessagePOJO eventMessagePOJO) throws RuntimeException {
+
+        //check for invalid inputs
```
Comment (mattclarke) [inline EventMessageVisualisationStreamer/src/eventmessage/EventMessagePOJOToEventMessage.java]: // Check for invalid inputs

File: EventMessageVisualisationStreamer/src/eventmessage/EventMessagePOJOToEventMessageTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    @org.junit.Test
+    public void convertPOJOWithDetectorsReturnsByteArray() {
+        eventMessagePOJO.addDetector(1);
+        eventMessagePOJO.addDetector(2);
+        eventMessagePOJO.addDetector(3);
+
+        byte[] result = EventMessagePOJOToEventMessage.convert(eventMessagePOJO);
+        //assertNotNull("Should not be null", result);
+    }
```
Comment (mattclarke) [inline EventMessageVisualisationStreamer/src/eventmessage/EventMessagePOJOToEventMessageTest.java]: Can we have an assert?

File: EventMessageVisualisationStreamer/src/test/java/EventMessage/EventMessagePOJOToEventMessageTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Created by STFC,ISIS on 07/06/2017.
```
Comment (mattclarke) [inline EventMessageVisualisationStreamer/src/test/java/EventMessage/EventMessagePOJOToEventMessageTest.java]: Change order

File: EventMessageVisualisationStreamer/src/test/java/EventMessage/EventMessagePOJOToEventMessageTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Before
+    public void setUp() throws Exception {
+        eventMessagePOJO = new EventMessagePOJO(DEFAULT_MESSAGE_ID, DEFAULT_PULSE_TIME);
+    }
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @org.junit.Test
```
Comment (mattclarke) [inline EventMessageVisualisationStreamer/src/test/java/EventMessage/EventMessagePOJOToEventMessageTest.java]: @Test

File: EventMessageVisualisationStreamer/src/test/java/EventMessage/EventMessagePOJOToEventMessageTest.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        byte[] eventMessage = EventMessagePOJOToEventMessage.convert(eventMessagePOJO);
+        EventMessagePOJO result = EventMessageToEventMessagePOJO.convert(eventMessage);
+
+        assertEquals(2, result.getDetectors().size());
+        assertEquals(1,result.getDetector(0));
+        assertEquals(2,result.getDetector(1));
+    }
+
+
+}
```
Comment (mattclarke) [inline EventMessageVisualisationStreamer/src/test/java/EventMessage/EventMessagePOJOToEventMessageTest.java]: Need blank line

**Comment by DChambersSTFC** [comment]
Requested changes completed. @mattclarke please update .gitignore and clean up files

File: EventMessageVisualisationStreamer/src/main/java/EventMessage/EventMessagePOJO.java [hunk 6]
```diff
@@ -1,9 +1,10 @@
-package eventmessage;
+package EventMessage;
 
 import java.util.ArrayList;
 
 /**
  * POJO for the Event Message, designed for easier use when not being sent over Kafka.
+ * Created by ISIS,STFC on 07/06/2017.
```
Comment (mattclarke) [inline EventMessageVisualisationStreamer/src/main/java/EventMessage/EventMessagePOJO.java]: space after comma

File: EventMessageVisualisationStreamer/src/main/java/EventMessage/EventMessagePOJOToEventMessage.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            EventMessage.addPulseTime(builder, eventMessagePOJO.getPulseTime());
+            // Convert to byte array and return
+            int eventMessage = EventMessage.endEventMessage(builder);
+            builder.finish(eventMessage);
+            return builder.sizedByteArray();
+        }
+    }
+
+
+}
```
Comment (mattclarke) [inline EventMessageVisualisationStreamer/src/main/java/EventMessage/EventMessagePOJOToEventMessage.java]: new line at end

**Pull request merged**